src/Compilers/CSharp/Portable/CodeGen/CodeGenerator.cs

```diff
@@ -102,12 +131,17 @@ private enum IndirectReturnState : byte
 
             _boundBody = Optimizer.Optimize(
                 boundBody, 
-                debugFriendly: _optimizations != OptimizationLevel.Release, 
+                debugFriendly: _ilEmitStyle != ILEmitStyle.Release, 
                 stackLocals: out _stackLocals);
 
             _methodBodySyntaxOpt = (method as SourceMethodSymbol)?.BodySyntax;
         }
 
+        private bool IsDebugPlus()
+        {
+            return this._module.Compilation.Options.DebugPlusMode;
```

Comment:
I think we can remove extendedCustomDebugInformation, I don't see the value of having it now.
Why isn't OptimizationLevel.DebugFriendlyRelease appropriate?